### PR TITLE
Indirect - JumpFit - Ensure output workspace is created and plotted in the interface

### LINKED ISSUE #21204: JumpFit - Plot Result and Plot Guess produce error messages
In the Interfaces > Indirect > Data Analysis > JumpFit interface, both plot result and plot guess produce errors (the mini-plot is therefore not updated). 

These errors follow from the result workspace not being created and added to the ADS.

### PR DESCRIPTION
**Pull request by brandonhewer**
Currently, in the Interfaces > Indirect > Data Analysis > JumpFit interface, running the routine from this interface, does not create an output workspace, which results in an error being thrown when attempting to plot. This occurs as a result of the output workspace property not being set to the correct value when creating the fit algorithm.

This PR fixes this issue, by ensuring the workspace is created.

**To test:**
1. Navigate to Interfaces > Indirect > Data Analysis > JumpFit.
2. Provide the supplied test file in the 'Sample' field.
3. Select any fit function.
4. Click 'Run'.
5. Ensure the '*_Workspace', '*_Parameters' are created and the fit and difference are added to the preview plots. Also ensure the fitting parameters are updated in the property table.

Fixes #21204 . 

**Test Files**
[irs26176_graphite002_conv_1LFixF_s0_to_9_Result.zip](https://github.com/mantidproject/mantid/files/1518441/irs26176_graphite002_conv_1LFixF_s0_to_9_Result.zip)

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/JumpFit.cpp [hunk 1]
```diff
@@ -139,19 +140,18 @@ void JumpFit::fitAlgDone(bool error) {
     return;
   m_uiForm.pbPlot->setEnabled(true);
   m_uiForm.pbSave->setEnabled(true);
-  std::string outName = m_fitAlg->getPropertyValue("Output");
 
   // Get output workspace name
-  std::string outWsName = outName + "_Workspace";
+  std::string outWsName = m_baseName + "_Workspace";
```
Comment (ElliotAOram) [inline qt/scientific_interfaces/Indirect/JumpFit.cpp]: nit: Can this be `const auto`?

File: qt/scientific_interfaces/Indirect/JumpFit.cpp [hunk 2]
```diff
@@ -457,39 +458,11 @@ IFunction_sptr JumpFit::createFunction(const QString &functionName) {
...
-    deleteAlg->execute();
-  }
-}
-
 /**
  * Handles mantid plotting
  */
 void JumpFit::plotClicked() {
-  std::string outWsName = m_fitAlg->getPropertyValue("Output") + "_Workspace";
+  std::string outWsName = m_baseName + "_Workspace";
```
Comment (ElliotAOram) [inline qt/scientific_interfaces/Indirect/JumpFit.cpp]: const auto

File: qt/scientific_interfaces/Indirect/JumpFit.cpp [hunk 3]
```diff
@@ -498,7 +471,7 @@ void JumpFit::plotClicked() {
  * Handles saving of workspace
  */
 void JumpFit::saveClicked() {
-  std::string outWsName = m_fitAlg->getPropertyValue("Output") + "_Workspace";
+  std::string outWsName = m_baseName + "_Workspace";
```
Comment (ElliotAOram) [inline qt/scientific_interfaces/Indirect/JumpFit.cpp]: const auto

File: qt/scientific_interfaces/Indirect/JumpFit.cpp [hunk 4]
```diff
@@ -139,19 +140,18 @@ void JumpFit::fitAlgDone(bool error) {
...
 
   // Get output workspace name
-  std::string outWsName = outName + "_Workspace";
+  std::string outWsName = m_baseName + "_Workspace";
   IndirectDataAnalysisTab::updatePlot(outWsName, m_uiForm.ppPlotTop,
                                       m_uiForm.ppPlotBottom);
-
   // Update parameters in UI
-  std::string paramTableName = outName + "_Parameters";
+  std::string paramTableName = m_baseName + "_Parameters";
```
Comment (ElliotAOram) [inline qt/scientific_interfaces/Indirect/JumpFit.cpp]: const auto

**Comment by ElliotAOram** [comment]
Functionally, changes are all good (other than some `const auto`s). Something to note that was found while testing:
* When you load the data in initially, the Q range in the property table is updated, but the miniplot is not.
This however is probably best resolved in a different issue

**Comment by brandonhewer** [comment]
@ElliotAOram See PR #21304 - This moves all similar functionality of fit analysis tabs to one place and fixes the mentioned issue.

**Review by ElliotAOram — APPROVED** [review]
Thanks @brandonhewer. This should pass builds as requested changes are simple. Assuming it does, this is ready to go in

**Pull request merged**